examples/minwebgl/video_as_texture/src/main.rs

```diff
@@ -0,0 +1,101 @@
+use minwebgl::{self as gl, JsCast};
+
+fn main()
+{
+  gl::spawn_local( async move { run().await.unwrap() } );
+}
+
+async fn run() -> Result< (), gl::WebglError >
+{
+  gl::browser::setup( Default::default() );
+  let gl = gl::context::retrieve_or_make()?;
+
+  let vertex_shader_src = include_str!( "../shaders/main.vert" );
+  let fragment_shader_src = include_str!( "../shaders/main.frag" );
+  let program = gl::ProgramFromSources::new( vertex_shader_src, fragment_shader_src ).compile_and_link( &gl )?;
+  gl.use_program( Some( &program ) );
+
+  let path = "static/simon's_cat.mp4";
+  let video_width = 640;
+  let video_height = 480;
+
+  let video_element = load_video( path, video_width, video_height )
+  .await
+  .expect( "Failed to load video" );
+  let texture = gl::texture::video::setup( &gl ).expect( "Failed to setup texture" );
+
+  let data : [ f32; 16 ] =
+  [//x     y     t_x   t_y
+    -0.3,  0.5,  0.0,  0.0,
+     0.3,  0.5,  1.0,  0.0,
+    -0.3, -0.5,  0.0,  1.0,
+     0.3, -0.5,  1.0,  1.0,
+  ];
+
+  let position_attribute_location = 0;
+  let tex_coord_attribute_location = 1;
+
+  let data_buffer = gl::buffer::create( &gl )?;
+  gl::buffer::upload( &gl, &data_buffer, &data, gl::GL::STATIC_DRAW );
+  gl::BufferDescriptor::new::< [ f32; 2 ] >()
+  .stride( 4 )
+  .offset( 0 )
+  .attribute_pointer( &gl, position_attribute_location, &data_buffer )?;
+  gl::BufferDescriptor::new::< [ f32; 2 ] >()
+  .stride( 4 )
+  .offset( 2 )
+  .attribute_pointer( &gl, tex_coord_attribute_location, &data_buffer )?;

```

Comment:
This buffer setup is unnecessary. You can define this arrays in the shader, and use glVertexIndex to get the value